hw06-atm/src/main/java/ru/otus/homework/ATM.java

```diff
@@ -0,0 +1,138 @@
+package ru.otus.homework;
+
+import lombok.extern.java.Log;
+import org.apache.commons.collections4.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+/**
+ * Class for ATM, which holds a number of Cell(s) with default values or custom ones
+ */
+@Log
+public class ATM {
+    private Comparator<Cell> compareCells = Comparator.comparing(Cell::getNominal, Comparator.reverseOrder());
```

Comment:
Если что-то можно сделать final, то лучше делать его final.
```suggestion
    private final Comparator<Cell> compareCells = Comparator.comparing(Cell::getNominal, Comparator.reverseOrder());
```
В других местах сами посмотрите на предмет final.